### PR TITLE
Improve Carousel performance

### PR DESCRIPTION
**Pull request by alcferreira**
ISSUES CLOSED: #299

### CODE REVIEW

File: packages/carousel/src/styledComponents/Carousel.js [hunk 1]
```diff
@@ -19,32 +31,28 @@ const Container = styled.div`
...
+    transition: transform ${props => props.itemConfig.transitionTime}s;
+    will-change: transform;
+
     li {
       margin: ${props => props.itemConfig.margin}px;
-      transition: all ${props => props.itemConfig.transitionTime}s;
       width: ${props => props.itemConfig.width}px;
       height: ${props => props.itemConfig.height}px;
 
+      transition: all ${props => props.itemConfig.transitionTime}s;
```
Comment (vNakamura) [inline packages/carousel/src/styledComponents/Carousel.js]: Maybe we'll get a little performance improvement if we set the transition only for the objects that we change (`transform` and `margin`)

File: packages/carousel/src/Carousel.js [hunk 2]
```diff
@@ -27,31 +27,53 @@ class Carousel extends React.Component {
...
+    }
   };
 
   render() {
     const { imageSet, itemConfig } = this.props;
     return (
       <Container
         activeItem={this.state.activeItem}
-        itemConfig={{ ...defaultConfig, ...itemConfig }}
+        itemConfig={{ ...itemConfig, ...defaultConfig }}
```
[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/carousel/src/Carousel.js]: Yeah! Now that you pointed out it, this change makes no sense... 
Default config should be the first argument... I'll drop this commit and dig into it to see what was going on

File: packages/carousel/src/styledComponents/Carousel.js (hunk 1 above)
Comment (alcferreira) [reply packages/carousel/src/styledComponents/Carousel.js]: Good catch! It was changing opacity too. I'll change it

File: packages/carousel/src/Carousel.js [hunk 3]
```diff
@@ -88,7 +88,7 @@ class Carousel extends React.Component {
       width: PropTypes.number,
       height: PropTypes.number,
       margin: PropTypes.number,
-      fontSize: PropTypes.string,
+      fontSize: PropTypes.number,
```
Comment (fczuardi) [inline packages/carousel/src/Carousel.js]: this is a breaking change, please inform this on the commit message

File: packages/carousel/src/Carousel.js [hunk 4]
```diff
@@ -27,23 +27,44 @@ class Carousel extends React.Component {
...
     this.props.onChange(activeItem);
     this.setState({ activeItem });
   };
 
   componentDidMount = () => {
+    this.setInterval();
+  };
+
+  componentWillUnmount = () => {
+    this.clearInterval();
```
Comment (fczuardi) [inline packages/carousel/src/Carousel.js]: :+1:

File: packages/carousel/src/Carousel.js [hunk 5]
```diff
@@ -65,7 +87,7 @@ class Carousel extends React.Component {
                   width="100%"
                   height="100%"
                 />
-                <Text className="caption" title>
+                <Text size={itemConfig.fontSize} align="center" title>
```
Comment (fczuardi) [inline packages/carousel/src/Carousel.js]: :+1:

File: packages/carousel/src/Carousel.js (hunk 3 above)
Comment (alcferreira) [reply packages/carousel/src/Carousel.js]: I'll make a rebase on the commit

File: packages/carousel/src/Carousel.test.js [hunk 6]
```diff
@@ -63,7 +63,7 @@ describe("Carousel", () => {
     expect(setIntervalSpy).toHaveBeenCalledTimes(1);
   });
 
-  test("clearInterval should be cleared on unmount", () => {
+  test("clearInterval should be called on unmount", () => {
```
Comment (fczuardi) [inline packages/carousel/src/Carousel.test.js]: :smile:

**Pull request merged**